runtime/nvqir/cudensitymat/CMakeLists.txt

```diff
@@ -51,53 +51,73 @@ endif()
 message(STATUS "CUDENSITYMAT_INC: ${CUDENSITYMAT_INC}")
 message(STATUS "CUDENSITYMAT_LIB: ${CUDENSITYMAT_LIB}")
 
-get_filename_component(CUDENSITYMAT_INCLUDE_DIR ${CUDENSITYMAT_INC} DIRECTORY)
-get_filename_component(CUDENSITYMAT_LIB_DIR ${CUDENSITYMAT_LIB} DIRECTORY)
-get_filename_component(CUTENSOR_LIB_DIR ${CUTENSOR_LIB} DIRECTORY)
-SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_RPATH}:${CUDENSITYMAT_LIB_DIR}:${CUTENSOR_LIB_DIR}")
-
-add_library(${LIBRARY_NAME} SHARED
-  CuDensityMatSim.cpp
-  mpi_support.cpp
-  CuDensityMatTimeStepper.cpp
-  RungeKuttaIntegrator.cpp
-  CuDensityMatExpectation.cpp
-  CuDensityMatEvolution.cpp
-  CuDensityMatState.cpp
-  CuDensityMatContext.cpp
-  CuDensityMatOpConverter.cpp
-  CuDensityMatUtils.cpp
-)
+# Determine cudensitymat version
```

Comment:
I suggest to split this PR into two - one for the qemu elimination/pipeline changes, and a separate one for the cuquantum related changes.